pkg/dockerregistry/server/quotarestrictedblobstore.go

```diff
@@ -120,9 +121,52 @@ func (bw *quotaRestrictedBlobWriter) Commit(ctx context.Context, provisional dis
 	return can, err
 }
 
-// admitBlobWrite checks whether the blob does not exceed image quota, if set. Returns
+// admitBlobWrite checks whether the blob does not exceed image quota or limit ranges, if set. Returns
 // ErrAccessDenied error if the quota is exceeded.
-func admitBlobWrite(ctx context.Context, repo *repository) error {
+func admitBlobWrite(ctx context.Context, repo *repository, size int64) error {
+	if err := admitLimitRanges(ctx, repo, size); err != nil {
+		return err
+	}
+
+	if err := admitQuotas(ctx, repo); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// admitLimitRanges checks the blob against any established limit ranges.
+func admitLimitRanges(ctx context.Context, repo *repository, size int64) error {
+	if size < 1 {
+		return nil
+	}
+
+	limitranges, err := repo.limitClient.LimitRanges(repo.namespace).List(kapi.ListOptions{})
+	if err != nil {
+		if kerrors.IsForbidden(err) {
```

Comment:
Why? I agree with @liggitt idea, having this on by default should be the way to go and flag `--disable-quota-limits` should be used with `oadm registry` to disable it explicitly. The other thing is existing deployments that should allow modifying it, this is where the `OPENSHIFT_ENFORCE_QUOTA` should be used. Generally the flag would set that env to `0` if specified. All other cases we should enforce quota.